# HG changeset patch
# Parent 79327fd9e5a410670b0b081b08a98a0e2375e3b6
# User Ben Kelly <ben@wanderview.com>
Fix cache tests not to leak prefs.


diff --git a/dom/tests/mochitest/cache/test_cache.html b/dom/tests/mochitest/cache/test_cache.html
--- a/dom/tests/mochitest/cache/test_cache.html
+++ b/dom/tests/mochitest/cache/test_cache.html
@@ -7,33 +7,32 @@
   <script type="text/javascript" src="/tests/SimpleTest/SimpleTest.js"></script>
   <link rel="stylesheet" type="text/css" href="/tests/SimpleTest/test.css" />
   <script type="text/javascript" src="worker_driver.js"></script>
 </head>
 <body>
 <iframe id="frame"></iframe>
 <script class="testbody" type="text/javascript">
 SimpleTest.waitForExplicitFinish();
-SpecialPowers.setBoolPref("dom.window-caches.enabled", true);
 SpecialPowers.pushPrefEnv({
   "set": [["dom.window-caches.enabled", true],
+          ["dom.worker-caches.enabled", true],
           ["dom.fetch.enabled", true]]
 }, function() {
   var frame = document.getElementById("frame");
   frame.src = "test_cache_frame.html";
   frame.onload = function() {
     var contentWindow = frame.contentWindow;
 
     addEventListener("message", function(evt) {
       ok(evt.data.success, "frame should have succeeded");
       frame.src = "about:blank";
       frame.parentNode.removeChild(frame);
       frame = null;
 
-      SpecialPowers.setBoolPref("dom.worker-caches.enabled", true);
       workerTestExec("test_cache.js");
     });
 
     contentWindow.postMessage({
       type: "start"
     }, "*");
   }
 });
